contents/tutorials/time-on-page.md

```diff
@@ -0,0 +1,224 @@
+---
+title: How to calculate time on page
+date: 2023-08-01
+author: ["ian-vanagas"]
+showTitle: true
+sidebar: Docs
+featuredImage: ../images/tutorials/banners/tutorial-14.png
+tags: ['hogql', 'insights']
+---
+
+Understanding how users spend their time on your site helps you understand your site’s strengths and weaknesses. Calculating the time on page is the key metric for doing this. In this tutorial, we show you how to calculate time on page and related metrics using PostHog.
+
+The challenge with time on page is measuring activity. Because of the event-driven structure of PostHog, we can’t be 100% sure a session is active at any point. We estimate by getting the time between a first pageview event and a subsequent pageview or pageleave event, filtering out extreme values, and averaging these together.
```

Comment:
```suggestion
The challenge with time on page is measuring activity. Because of the event-driven structure of PostHog, we can’t be 100% sure a session is active at any point. Thus we can estimate time spent by getting the time between a first pageview event and a subsequent pageview or pageleave event, filtering out extreme values, and averaging these together.
```